### ISSUE TITLE
Issue with eslint-import-resolver -> handleRemainingUrl

### ISSUE THREAD

**Issue by prisis** [jsenv/core#379]
Hey :), 

can be that i found a bug with this function (https://github.com/jsenv/core/blob/main/packages/independent/eslint-import-resolver/src/resolver.js#L279).

Currently this function returns null for a path that triggers a error inside the "eslint-module-utils/resolve" on line 135 - 140

```js
    const resolved = withResolver(resolver, config); 
	// -> {
	//  found: true,
	//  path: null,
	// }

    if (!resolved.found) { continue; }
    // else, counts
    cache(resolved.path);
```

Because of the info that a path was found, but the path key, it self is null you triggering a error inside the "import/no-useless-path-segments" rule on line 122  when "eslint --fix" is run, because the check on `if (resolvedPath === undefined) {` is skipped.

From my view, the function "handleRemainingUrl" should look like

```js
const handleRemainingUrl = (url) => {
  return {
    found: true,
    path: url,
  };
};
```

**Comment by dmail** [comment]
You are right, I'm opening a PR to fix right now

**Comment by ljharb** [comment]
@dmail i'm pretty confused why there's a vendored copy of eslint-import-resolver in here; can you help me understand?

**Comment by dmail** [comment]
@ljharb what is `eslint-import-resolver` you are refering to? 

Because the package [@jsenv/eslint-import-resolver](https://github.com/jsenv/core/tree/main/packages/independent/eslint-import-resolver) is not a vendored copy of something. It is an import resolver meant to be used with eslint-plugin-import.

It exists mainly to provide node esm resolution to ESLint via eslint-plugin-import.

**Comment by ljharb** [comment]
ahhh ok, then i was just confused, thanks for clarifying :-)

**Comment by dmail** [comment]
@prisis I just found a test case explaining why code returns `null`:

https://github.com/jsenv/core/blob/95ab466d95a87e5b5e7e7b007a022a40dc9772f3/packages/independent/eslint-import-resolver/tests/resolve.test.mjs#L284-L288

This is in case there is an import using `http:`, to prevent eslint-plugin-import to try to read a file when it's an http url.

So I am wondering what is the import specifier you want to resolve and expect to pass trough? Because it looks like I have to keep returning null for http urls

**Comment by dmail** [comment]
@ljharb Happy that I have clarified my position but I'm curious to understand what caused confusion on your side?

**Comment by ljharb** [comment]
@dmail just jet lag and dividing my attention, i suspect :-) sorry for the noise

**Comment by prisis** [comment]
I can provide a repoduction,

Clone: https://github.com/anolilab/javascript-style-guide
Run: pnpm i
Change line: https://github.com/anolilab/javascript-style-guide/blob/main/packages/eslint-config/src/config/plugins/import.ts#L295 

From  `"import/no-useless-path-segments": ["error", { commonjs: false, noUselessIndex: true }],`
To `"import/no-useless-path-segments": ["error", { commonjs: true, noUselessIndex: true }],`

Run: `pnpm run build:packages`
Run: `pnpm run lint:eslint:fix`

Then you will see error, the error is happening inside https://github.com/anolilab/javascript-style-guide/blob/main/packages/eslint-config/bin/generate-eslint-cofig.js#L3

**Comment by dmail** [comment]
Thank you for the reproduction, I have the fix, more info into the upcoming pull request

**Comment by dmail** [comment]
I have published `@jsenv/eslint-import-resolver@8.0.3` on NPM, it contains the fix you need. Thanks for opening the issue.

**Comment by prisis** [comment]
Thank you!
